pkg/kubelet/kubelet_pods.go

```diff
@@ -1470,12 +1470,12 @@ func (kl *Kubelet) generateAPIPodStatus(pod *v1.Pod, podStatus *kubecontainer.Po
 	}
 
 	// pods are not allowed to transition out of terminal phases
-	if pod.Status.Phase == v1.PodFailed || pod.Status.Phase == v1.PodSucceeded {
+	if oldPodStatus.Phase == v1.PodFailed || oldPodStatus.Phase == v1.PodSucceeded {
```

Comment:
we may actually have to check both the cached and apiserver status.  If either are terminal we must not transition.  so this whole if might need to be duplicated.  oldPodStatus is *either* kubelets internal or apis historical, so we probably have to look at both (or both if we had cached status)

Reply:
Duplicated

Reply:
the other bug we are seeing is that when we disallow a transition we shouldn’t set reason or message.  we also need to check for illegal transition after synchandler runs, since they are allowed to change phase too.